### PR TITLE
[Ebook Words] Create list (issue #11)

### LINKED ISSUE #43: [Code] Linters
- [x] add Eslint rules (airbnb)
- [x] add Prettier config
- [x] add husky precommit ??
- [x] add stylelint ??

### PR DESCRIPTION
**Pull request by kastrubait**
Implemented:
- [x] the home page of the tutorial (issue https://github.com/whattablackhole/rslang/issues/35)
- [x] collections page (issue https://github.com/whattablackhole/rslang/issues/36)
- [ ] Create list of words
- [x] Add rule airbnb-base, stylelint, husky,
- [x] Eslint bugs fixed

#### With the current eslint settings, it is not possible to fulfill the component naming convention!!

### CODE REVIEW

File: rslang/src/app/ebook/models/wordStudy.model.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Word } from '../../shared/models/word.model';
+
+export interface WordStady extends Word {
```
Comment (whattablackhole) [inline rslang/src/app/ebook/models/wordStudy.model.ts]: wrong name

File: rslang/src/app/ebook/ebook.module.ts [hunk 2]
```diff
@@ -1,10 +1,13 @@
 import { NgModule } from '@angular/core';
-import { CommonModule } from '@angular/common';
-import { EbookPageComponent } from './components/ebook-page/ebook-page.component';
+import { HttpClientModule } from '@angular/common/http';
+import { SharedModule } from '../shared/shared.module';
+
+import { EbookHomeComponent } from './components/ebook-page/ebook-home.component';
```
Comment (whattablackhole) [inline rslang/src/app/ebook/ebook.module.ts]: remove  name part 'Component' from the word, add eslint rule if needed

Comment (kastrubait) [reply rslang/src/app/ebook/ebook.module.ts]: The component was named so in the develop, so I thought it necessary to leave it that way))

File: rslang/src/app/ebook/components/collections/collections.component.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<section class="collections">
+  <ng-container *ngIf="collections.length > 0; else isEmptyBlock">
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/collections/collections.component.html]: can be simplified to "collections.length"

File: rslang/src/app/ebook/components/collections/collections.component.html [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        class="collections-container--item"
+        *ngFor="let collection of collections"
+        (click)="setGroupSelected(collection)">
+        <h2>{{ collection.name }}</h2>
+        <span>{{ collection.progress }}</span>
+      </div>
+    </div>
+  </ng-container>
+  <ng-template #isEmptyBlock>
+    <div  class="collections-container--empty">
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/collections/collections.component.html]: unnecessary space

File: rslang/src/app/ebook/components/collections/collections.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CollectionsComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/collections/collections.component.spec.ts]: Maybe toBeDefined?

File: rslang/src/app/ebook/components/collections/collections.component.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+
+import { Collection } from '../../models/collection.models';
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/collections/collections.component.ts]: file should be named collection.model as soon as you should have only ONE model in it

File: rslang/src/app/ebook/components/collections/collections.component.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  selector: 'app-collections',
+  templateUrl: './collections.component.html',
+  styleUrls: ['./collections.component.scss'],
+})
+export class CollectionsComponent {
+  @Input() collections!: Collection[];
+  @Output() group: EventEmitter<number> = new EventEmitter<number>();
+
+  constructor() {
+    console.log(this.collections);
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/collections/collections.component.ts]: ?

File: rslang/src/app/ebook/components/ebook-page/ebook-home.component.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  collectionWord!: Word[];
+
+  group = 0;
+  wordSubscription!: Subscription;
+
+  constructor(private wordsDataService: WordsDataService) {}
+
+  ngOnInit(): void {
+    this.collectionWord = this.wordsDataService.GetWords();
+    console.log(this.collections);
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/ebook-page/ebook-home.component.ts]: ?

File: rslang/src/app/ebook/components/ebook-page/ebook-home.component.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  selector: 'app-ebook-home',
+  templateUrl: './ebook-home.component.html',
+  styleUrls: ['./ebook-home.component.scss'],
+})
+export class EbookHomeComponent implements OnInit {
+  collections: Collection[] = CONFIG_EBOOK.collections;
+  collectionWord!: Word[];
+
+  group = 0;
+  wordSubscription!: Subscription;
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/ebook-page/ebook-home.component.ts]: unused property

File: rslang/src/app/ebook/components/ebook-page/ebook-home.component.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+@Component({
+  selector: 'app-ebook-home',
+  templateUrl: './ebook-home.component.html',
+  styleUrls: ['./ebook-home.component.scss'],
+})
+export class EbookHomeComponent implements OnInit {
+  collections: Collection[] = CONFIG_EBOOK.collections;
+  collectionWord!: Word[];
+
+  group = 0;
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/ebook-page/ebook-home.component.ts]: unused

File: rslang/src/app/ebook/components/list-words/list-words.component.spec.ts [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ListWordsComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/list-words/list-words.component.spec.ts]: defined?

File: rslang/src/app/ebook/ebook.module.ts (hunk 2 above)
Comment (Nikitzu) [reply rslang/src/app/ebook/ebook.module.ts]: No it is not

File: rslang/src/app/ebook/models/collection.models.ts [hunk 12]
```diff
@@ -0,0 +1,6 @@
+export interface Collection {
```
Comment (Nikitzu) [inline rslang/src/app/ebook/models/collection.models.ts]: Unclear name of the model. Collection? Collection of what?

File: rslang/src/app/ebook/models/wordStudy.model.ts (hunk 1 above)
Comment (Nikitzu) [reply rslang/src/app/ebook/models/wordStudy.model.ts]: agree

File: rslang/src/app/ebook/models/wordStudy.model.ts [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+    };
+    usedInSavannah: {
+      total: number;
+      successfully: number;
+    };
+    usedInAudiocall: {
+      total: number;
+      successfully: number;
+    };
+  };
```
Comment (Nikitzu) [inline rslang/src/app/ebook/models/wordStudy.model.ts]: toStudy, usedInSprint, usedInSavannah, usedInAudiocall - should have their own models. All objects should have models if they are custom typed. By the way three of four properties will have the same type that should be named somehow with respect of that fact.

File: rslang/src/app/shared/shared.module.ts [hunk 14]
```diff
@@ -1,9 +1,13 @@
...
+import { HttpClientModule } from '@angular/common/http';
+
 import { NotFoundPageComponent } from './components/not-found-page/not-found-page.component';
 
 @NgModule({
   declarations: [NotFoundPageComponent],
-  imports: [CommonModule],
+  imports: [CommonModule, HttpClientModule],
+  exports: [CommonModule],
+  providers: [],
```
Comment (Nikitzu) [inline rslang/src/app/shared/shared.module.ts]: Empty?

File: rslang/src/app/shared/models/options.model.ts [hunk 15]
```diff
@@ -0,0 +1,9 @@
+import { HttpHeaders, HttpParams } from '@angular/common/http';
+import { Word } from './word.model';
+
+export interface Options {
+  body?: Word | Word[];
+  headers?: HttpHeaders | { [header: string]: string | string[] };
+  params?: HttpParams | { [param: string]: string | string[] };
```
Comment (Nikitzu) [inline rslang/src/app/shared/models/options.model.ts]: headers and params - should be typed instead of { [header: string]: string | string[] }

**Review by Nikitzu — CHANGES_REQUESTED** [review]

File: rslang/src/app/ebook/models/wordStudy.model.ts (hunk 13 above)
Comment (kastrubait) [reply rslang/src/app/ebook/models/wordStudy.model.ts]: fix

File: rslang/src/app/ebook/components/collections/collections.component.ts (hunk 6 above)
Comment (kastrubait) [reply rslang/src/app/ebook/components/collections/collections.component.ts]: fix

File: rslang/src/app/ebook/models/word-statistics.model.ts [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+
+interface Progress {
+  [propName: string]: Result[];
+}
+
+interface Result {
+  update: number;
+  total: number;
+  successfully: number;
+}
```
Comment (Nikitzu) [inline rslang/src/app/ebook/models/word-statistics.model.ts:8]: One file - one interface

**Review by Nikitzu — CHANGES_REQUESTED** [review]

File: rslang/src/app/ebook/components/collections/collections.component.ts [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+} from '@angular/core';
+
+import { CollectionWords } from '../../models/collection-words.model';
+
+@Component({
+  selector: 'app-collections',
+  templateUrl: './collections.component.html',
+  styleUrls: ['./collections.component.scss'],
+})
+export class Collections {
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/collections/collections.component.ts]: Can we define what collection do we mean?

File: rslang/src/app/ebook/components/ebook-page/ebook-home.component.ts [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+import { Word } from '../../../shared/models/word.model';
+import { CollectionWords } from '../../models/collection-words.model';
+
+@Component({
+  selector: 'app-ebook-home',
+  templateUrl: './ebook-home.component.html',
+  styleUrls: ['./ebook-home.component.scss'],
+})
+export class EbookHome implements OnInit {
+  collections: CollectionWords[] = CONFIG_EBOOK.collections;
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/ebook-page/ebook-home.component.ts]: WordsCollection?

File: rslang/src/app/ebook/components/list-words/list-words.component.ts [hunk 19]
```diff
@@ -0,0 +1,12 @@
...
+
+import { Word } from '../../../shared/models/word.model';
+
+@Component({
+  selector: 'app-list-words',
+  templateUrl: './list-words.component.html',
+  styleUrls: ['./list-words.component.scss'],
+})
+export class ListWords {
+  @Input() wordsList!: Word[];
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/list-words/list-words.component.ts]: can be just words as Interface

File: rslang/src/app/ebook/models/collection-words.model.ts [hunk 20]
```diff
@@ -0,0 +1,7 @@
+export interface CollectionWords {
```
Comment (Nikitzu) [inline rslang/src/app/ebook/models/collection-words.model.ts]: WordsCollection

File: rslang/src/app/shared/constants/api-url.ts [hunk 21]
```diff
@@ -0,0 +1,10 @@
+const BASE_URL = 'https://afternoon-falls-25894.herokuapp.com';
```
Comment (Anntol) [inline rslang/src/app/shared/constants/api-url.ts]: Please use https://rslang-team4-whattablackhole.netlify.app/

File: rslang/src/app/ebook/models/word-statistics.model.ts [hunk 22]
```diff
@@ -0,0 +1,17 @@
+import { Word } from '../../shared/models/word.model';
+
+export interface WordWithStatistics extends Word {
+  isDifficult: boolean;
+  isRemove: boolean;
+  toStudy: Progress;
+}
+
+interface Progress {
```
Comment (Nikitzu) [inline rslang/src/app/ebook/models/word-statistics.model.ts]: Progress of what? All progress or for some game?

File: rslang/src/app/ebook/models/word-statistics.model.ts [hunk 23]
```diff
@@ -0,0 +1,17 @@
...
+  [propName: string]: Result[];
+}
+
+interface Result {
```
Comment (Nikitzu) [inline rslang/src/app/ebook/models/word-statistics.model.ts]: Result of what? All result or for some game?

**Review by Nikitzu — CHANGES_REQUESTED** [review]

File: rslang/src/app/ebook/models/word-statistics.model.ts (hunk 22 above)
Comment (kastrubait) [reply rslang/src/app/ebook/models/word-statistics.model.ts]: Structure contains data for each game, so all progres.

Comment (kastrubait) [reply rslang/src/app/ebook/models/word-statistics.model.ts]: And track progress for a word!

File: rslang/src/app/ebook/models/word-statistics.model.ts (hunk 23 above)
Comment (kastrubait) [reply rslang/src/app/ebook/models/word-statistics.model.ts]: result is the result of learning one word.

File: rslang/src/app/shared/constants/api-url.ts (hunk 21 above)
Comment (kastrubait) [reply rslang/src/app/shared/constants/api-url.ts]: fix

File: rslang/src/app/ebook/components/collections/collections.component.ts (hunk 17 above)
Comment (kastrubait) [reply rslang/src/app/ebook/components/collections/collections.component.ts]: This is a list of all collections

File: rslang/src/app/ebook/components/ebook-page/ebook-home.component.ts (hunk 18 above)
Comment (kastrubait) [reply rslang/src/app/ebook/components/ebook-page/ebook-home.component.ts]: fix

File: rslang/src/app/ebook/models/collection-words.model.ts (hunk 20 above)
Comment (kastrubait) [reply rslang/src/app/ebook/models/collection-words.model.ts]: fix

File: rslang/src/app/ebook/models/word-statistics.model.ts (hunk 16 above)
Comment (kastrubait) [reply rslang/src/app/ebook/models/word-statistics.model.ts:8]: ok

File: rslang/src/app/ebook/components/collections/words-collections.component.ts [hunk 24]
```diff
@@ -0,0 +1,19 @@
...
+import { WordsCollection } from '../../models/words-collection.model';
+
+@Component({
+  selector: 'app-words-collections',
+  templateUrl: './words-collections.component.html',
+  styleUrls: ['./words-collections.component.scss'],
+})
+export class WordsCollections {
+  @Input() wordsCollections!: WordsCollection[];
+  @Output() group: EventEmitter<number> = new EventEmitter<number>();
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/collections/words-collections.component.ts]: Name like an event. setGroup for example.

File: rslang/src/app/ebook/components/list-words/list-words.component.ts [hunk 25]
```diff
@@ -0,0 +1,12 @@
+import { Component, Input } from '@angular/core';
+
+import { Word } from '../../../shared/models/word.model';
+
+@Component({
+  selector: 'app-list-words',
+  templateUrl: './list-words.component.html',
+  styleUrls: ['./list-words.component.scss'],
+})
+export class ListWords {
```
Comment (Nikitzu) [inline rslang/src/app/ebook/components/list-words/list-words.component.ts]: WordsList?

**Review by Nikitzu — CHANGES_REQUESTED** [review]

**Review by Nikitzu — APPROVED** [review]

**Pull request merged**